arch/arm64/core/cache.c

```diff
@@ -39,20 +39,43 @@
 	__asm__ volatile ("dc " op ", %0" :: "r" (val) : "memory");	\
 })
 
-int arch_dcache_flush(void *addr, size_t size);
-int arch_dcache_invd(void *addr, size_t size);
-
 static size_t dcache_line_size;
 
+/*
+ * operation for data cache by virtual address to PoC
+ * ops:  K_CACHE_INVD: invalidate
+ *	 K_CACHE_WB: clean
+ *	 K_CACHE_WB_INVD: clean and invalidate
+ */
 int arch_dcache_range(void *addr, size_t size, int op)
 {
-	if (op == K_CACHE_INVD) {
-		arch_dcache_invd(addr, size);
-	} else if (op == K_CACHE_WB_INVD) {
-		arch_dcache_flush(addr, size);
-	} else {
+	uint64_t ctr_el0;
+	uint32_t dminline;
+	size_t line_size;
+	uintptr_t start_addr = (uintptr_t)addr;
+	uintptr_t end_addr = start_addr + size;
+
+	if (op != K_CACHE_INVD && op != K_CACHE_WB && op != K_CACHE_WB_INVD)
 		return -ENOTSUP;
-	}
+
+	ctr_el0 = read_sysreg(CTR_EL0);
+	dminline = (ctr_el0 >> CTR_EL0_DMINLINE_SHIFT) & CTR_EL0_DMINLINE_MASK;
+	line_size = 4 << dminline;
```

Comment:
Why not directly using `arch_dcache_line_size_get()`?

Reply:
I am not sure why use the following code to get line size in arch_dcache_line_size_get()?

dcache_line_size = cwg ? 4 << cwg : 4 << dminline;

I find it use 4 << DminLine to get the mininual line size in  sample code of "Cleaning by Virtual Address" in cortex a-v8 programer guider.

Reply:
@MrVan , could you please address why use cwg field to get dcache line size, I found in the other SW component such as tf-a and uboot, and also in ARM's sample code, it only get the minimal dcache line size by using dminline ? thanks.

Reply:
Indeed not clear what the CWG is for there.

Reply:
Yes, I removed CSW and use dminline only.

Reply:
Typo, removed CWG.